pkgs/tools/misc/shelldap/default.nix

```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchurl, perlPackages }:
+perlPackages.buildPerlPackage rec {
+  name = "shelldap-${version}";
+  version = "1.4.0";
+  src = fetchurl {
+    url = "https://bitbucket.org/mahlon/shelldap/downloads/shelldap-${version}.tar.gz";
+    sha256 = "07gkvvxcgw3pgkfy8p9mmidakciaq1rsq5zhmdqd8zcwgqkrr24i";
+  };
+  buildInputs = with perlPackages; [ perl YAMLSyck NetLDAP AlgorithmDiff IOSocketSSL AuthenSASL TermReadLineGnu TermShell ];
+  prePatch = ''
+    touch Makefile.PL
+  '';
+  installPhase = ''
+    runHook preInstall
+    install -D shelldap $out/bin/shelldap
```

Comment:
We need ```-m755``` to make it executable.

The whole thing: ```install -Dm755 -t $out/bin shellldap```

Reply:
Why is it required if the file was already executable before? When I list it it is actually executable:

    -r-xr-xr-x 1 me me 62748 janv.  1  1970 result/bin/shelldap

Also, why 755 and not 555? I though the nix store should be readonly? For what I tried it's 555 for the other programs:

    $ ls /nix/store/z5r5s8ccp7hlxkmaijhdikppcawa6d0p-gnutar-1.30/bin -l
    -r-xr-xr-x 1 leo leo 470240 janv.  1  1970 tar

Nevermind, I tried to set it to 755, and it goes back to 555 automatically. I guess nixos already prevent readable store, but I'd be curious to know why 755 is recommanded here.

I pushed the `-m755` version.

Reply:
> Nevermind, I tried to set it to 755, and it goes back to 555 automatically. I guess nixos already prevent readable store, but I'd be curious to know why 755 is recommanded here.

In the store, it's 555 and not 755, correct. The write bit makes no sense.

The reason for specifying the mode as part of the installation, is simply that some installs do not set the permissions properly, so if we're specific, we are less likely to hit problems down the road.

Reply:
Ok thanks. I corrected it back to 555.